syntax = "proto2";

package AdminPortProtocol;

message AddPluginParameters {
    required string id = 1;
    required string path = 2;
    repeated string config = 3;
}

message PluginParameters {
    repeated string plugins = 1;
}

message ShutdownParameters {
    optional string reason = 1;
}

message Request {
    enum RequestType {
        ADD_PLUGIN = 1;
        REMOVE_PLUGIN = 2;
        START_PLUGIN = 3;
        STOP_PLUGIN = 4;
        RESTART_PLUGIN = 5;
        RELOAD_PLUGIN = 6;
        SHUTDOWN = 255;
    }

    required RequestType type = 1;
    oneof data {
        AddPluginParameters add_plugin_params = 2;
        PluginParameters plugin_params = 3;
        ShutdownParameters shutdown_params = 4;
    }
}

message Response {
    enum ResponseType {
        OK = 1;
        ERROR = 2;
    }

    required ResponseType type = 1;
    required bool end = 2;
    optional int32 code = 3;
    optional string msg = 4;
}
